lnd_test.go

```diff
@@ -467,6 +467,211 @@ func testBasicChannelFunding(net *lntest.NetworkHarness, t *harnessTest) {
+	err = net.Bob.WaitForNetworkChannelOpen(ctxt, chanPoint2)
+	if err != nil {
+		t.Fatalf("bob didn't report channel: %v", err)
+	}
+	err = carol.WaitForNetworkChannelOpen(ctxt, chanPoint2)
+	if err != nil {
+		t.Fatalf("carol didn't report channel: %v", err)
+	}
+
+	// Update the fees for the channel Alice->Bob, and make sure
+	// all nodes learn about it.
+	const feeBase = 1000000
+	baseFee := int64(1500)
+	feeRate := int64(12)
+	timeLockDelta := uint32(66)
+
+	req := &lnrpc.PolicyUpdateRequest{
+		BaseFeeMsat:   baseFee,
+		FeeRate:       float64(feeRate),
+		TimeLockDelta: timeLockDelta,
+	}
+	req.Scope = &lnrpc.PolicyUpdateRequest_ChanPoint{
+		ChanPoint: chanPoint,
+	}
+
+	_, err = net.Alice.UpdateChannelPolicy(ctxb, req)
+	if err != nil {
+		t.Fatalf("unable to get alice's balance: %v", err)
+	}
+
+	// assertChannelPolicy asserts that the passed node's known channel
+	// policy for the passed chanPoint is consistent with Alice's current
+	// expected policy values.
+	assertChannelPolicy := func(node *lntest.HarnessNode,
+		chanPoint *lnrpc.ChannelPoint) {
+		fundingTxID, err := chainhash.NewHash(chanPoint.FundingTxid)
+		if err != nil {
+			t.Fatalf("could not get funding txid: %v", err)
+		}
+		cp := wire.OutPoint{
+			Hash:  *fundingTxID,
+			Index: chanPoint.OutputIndex,
+		}
+
+		// Get a DescribeGraph from the node.
+		descReq := &lnrpc.ChannelGraphRequest{}
+		chanGraph, err := node.DescribeGraph(ctxb, descReq)
+		if err != nil {
+			t.Fatalf("unable to query for alice's routing table: %v",
+				err)
+		}
+
+		edgeFound := false
+		for _, e := range chanGraph.Edges {
+			if e.ChanPoint == cp.String() {
+				edgeFound = true
+				if e.Node1Pub == net.Alice.PubKeyStr {
+					if e.Node1Policy.FeeBaseMsat != baseFee {
+						t.Fatalf("expected base fee "+
+							"%v, got %v", baseFee,
+							e.Node1Policy.FeeBaseMsat)
+					}
+					if e.Node1Policy.FeeRateMilliMsat != feeRate*feeBase {
+						t.Fatalf("expected fee rate "+
+							"%v, got %v", feeRate*feeBase,
+							e.Node1Policy.FeeRateMilliMsat)
+					}
+					if e.Node1Policy.TimeLockDelta != timeLockDelta {
+						t.Fatalf("expected time lock "+
+							"delta %v, got %v",
+							timeLockDelta,
+							e.Node1Policy.TimeLockDelta)
+					}
+				} else {
+					if e.Node2Policy.FeeBaseMsat != baseFee {
+						t.Fatalf("expected base fee "+
+							"%v, got %v", baseFee,
+							e.Node2Policy.FeeBaseMsat)
+					}
+					if e.Node2Policy.FeeRateMilliMsat != feeRate*feeBase {
+						t.Fatalf("expected fee rate "+
+							"%v, got %v", feeRate*feeBase,
+							e.Node2Policy.FeeRateMilliMsat)
+					}
+					if e.Node2Policy.TimeLockDelta != timeLockDelta {
+						t.Fatalf("expected time lock "+
+							"delta %v, got %v",
+							timeLockDelta,
+							e.Node2Policy.TimeLockDelta)
+					}
+				}
+			}
+		}
+
+		if !edgeFound {
+			t.Fatalf("did not find edge")
+		}
+
+	}
+
+	// Sleep to let the update propagate.
+	time.Sleep(300 * time.Millisecond)
```

Comment:
We can use the `SubcribeChannelGraph` call here instead. Utilizing this, we'll receive a notification once a new channel update is encountered.